src/Symmetry/AllowedAnisotropy.jl

```diff
@@ -61,38 +61,21 @@ function basis_for_symmetry_allowed_anisotropies(cryst::Crystal, i::Int; k::Int,
     D = unitary_irrep_for_rotation(R; N=2k+1)
     C = D * C
 
-    # Find an orthonormal basis for the columns of A, discarding linearly
-    # dependent columns.
-    C = colspace(C; atol=1e-12)
-
-    # It is tempting to sparsify here to make the ouput look nicer. Don't do
-    # this because (empirically) it is observed to significantly degrade
-    # accuracy in stevens_basis_for_symmetry_allowed_anisotropies().
-
-    # C = sparsify_columns(C; atol=1e-12)
-
-    return C
-end
-
-function stevens_basis_for_symmetry_allowed_anisotropies(cryst::Crystal, i::Int; k::Int, R::Mat3)
-    # Each column of C represents a coefficient vector c that can be contracted
-    # with spherical tensors T to realize an allowed anisotropy, Λ = cᵀ T.
-    C = basis_for_symmetry_allowed_anisotropies(cryst, i; k, R)
+    # Transform columns c of C to columns b of B such that 𝒜 = cᵀ T = bᵀ 𝒪.
+    # Effectively, this reexpresses the symmetry-allowed operators 𝒜 in the
+    # basis of Stevens operators 𝒪.
+    B = mapslices(C; dims=1) do c
+        transform_spherical_to_stevens_coefficients(k, c)
+    end
 
-    # Transform each column c to coefficients b that satisfy bᵀ 𝒪 = cᵀ T
-    B = [transform_spherical_to_stevens_coefficients(k, c) for c in eachcol(C)]
+    # If 𝒜 is symmetry-allowed, then its Hermitian and anti-Hermitian parts are
+    # independently symmetry-allowed. These are given by the real and imaginary
+    # parts of B. Create a real matrix with these two parts, and eliminate
+    # linearly-dependent vectors from the column space.
+    B = colspace(hcat(real(B), imag(B)); atol=1e-12)
```

Comment:
I guess this in some sense eliminates the imaginary part by construction in a slightly different way. It's not obvious to me what the effect of doing it this way is, but it seems to work.

Reply:
In the previous code, we were using an SVD to reduce the spanned vector space, and claiming "the result must be real". Although that seemed to be true in practice, it's not clear to me that SVD actually gives that guarantee.

In this PR, we're doing something more careful. The columns of B are complex vectors that span the space of symmetry-allowed operators, with no restriction on Hermiticity. However, we wish to apply an additional constraint that we only care about symmetry-allowed operators that are Hermitian. If we just take `real(B)`, we'd get symmetry-allowed operators that are Hermitian, but we're not guaranteed to get _all_ of them. Similarly, `imag(B)` will give a different space of symmetry-allowed operators, and if you multiply by `im`, then they are Hermitian also. I suspect that the combined columns `hcat(real(B), imag(B))` spans the full space of symmetry-allowed Hermitian operators. Then the call to `colspace` cleans up the redundancy.